### PR TITLE
Fix pytest deprecation warnings.

### PR DESCRIPTION
**Pull request by fdmalone**
The diff is huge because of reformatting. May be worth first merging #846.  But just fixed some deprecation warnings (numpy.product, numpy.math, converting size one array to scalar)  emitted by pytest.

### CODE REVIEW

File: src/openfermion/linalg/davidson_test.py [hunk 1]
```diff
@@ -116,7 +117,7 @@ def test_invalid_dimension(self):
 
 
 class DavidsonTest(unittest.TestCase):
-    """"Tests for Davidson class with a real matrix."""
+    """ "Tests for Davidson class with a real matrix."""
```
Comment (mpharrigan) [inline src/openfermion/linalg/davidson_test.py:120]: nit there's an extra quote here

Comment (fdmalone) [reply src/openfermion/linalg/davidson_test.py]: Fixed.

**Comment by fdmalone** [comment]
@mpharrigan needs approval

**Review by mpharrigan — APPROVED** [review]
thanks for the ping; I was waiting until I could see an actual manageable diff :)

**Pull request merged**